pysmt/solvers/cvc4.py

```diff
@@ -339,6 +339,13 @@ def walk_array_store(self, formula, args, **kwargs):
     def walk_array_select(self, formula, args, **kwargs):
         return self.mkExpr(CVC4.SELECT, args[0], args[1])
 
+    def walk_array_value(self, formula, args, **kwargs):
+        arr_type = self.env.stc.get_type(formula)
```

Comment:
Is this the easiest way to get the type of the array value? Isn't this information available directly from the FNode? 

(I know this is the same as done in MathSAT, just wondering if there is an opportunity to improve this)

Reply:
I just tested this locally, and you're right that: `arr_type = formula.get_type()` also works and seems easier. However, I noticed that every other solver file seems to use the env.stc approach (possibly by creating a new name for that function). I'm not sure if that's historical or if there's a particular reason?

Would you prefer to use the get_type from the FNode in this case?